DirectoryNotEmptyException when using mvn jetty:run-distro

**Description**

If I run `mvn jetty:run-distro` without first doing a `mvn clean` I get the following exception:
```
Caused by: java.nio.file.DirectoryNotEmptyException: /home/lachlan/webtide/src/.../target/jetty-base
    at sun.nio.fs.UnixFileSystemProvider.implDelete (UnixFileSystemProvider.java:247)
    at sun.nio.fs.AbstractFileSystemProvider.deleteIfExists (AbstractFileSystemProvider.java:110)
    at java.nio.file.Files.deleteIfExists (Files.java:1180)
    at org.eclipse.jetty.maven.plugin.JettyRunDistro.configureJettyBase (JettyRunDistro.java:344)
    at org.eclipse.jetty.maven.plugin.JettyRunDistro.startJetty (JettyRunDistro.java:240)
    ...
```

The javadoc for `Files.deleteIfExists()` says
> DirectoryNotEmptyException – if the file is a directory and could not otherwise be deleted because the directory is not empty

In commit 041a3260b6f04c1e709cf4f4b6fd41b558f5a9da this was changed from using `IO.delete(targetBase)` which recursively deletes directories, to using `Files.deleteIfExists()` which throws if the directory is not empty. From the look of the commit it looks like this was an unintentional behaviour change.

--- a/jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/JettyRunDistro.java
+++ b/jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/JettyRunDistro.java
@@ -341,7 +341,8 @@ public class JettyRunDistro extends JettyRunMojo
 
         targetBase = new File(target, "jetty-base");
         Path targetBasePath = targetBase.toPath();
-        Files.deleteIfExists(targetBase.toPath());
+        if (Files.exists(targetBasePath))
+            IO.delete(targetBase);
 
         targetBase.mkdirs();
 

